//  File: tao_admin_channel.proto
//  Author: Tom Roeder <tmroeder@google.com>
//
//  Description: Protocol buffers for TaoAdminChannel RPC.
//
//  Copyright (c) 2013, Google Inc.  All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package tao;

enum TaoAdminRequestType {
  TAO_ADMIN_RPC_UNKNOWN = 0;  // unused
  TAO_ADMIN_RPC_SHUTDOWN = 100;
  TAO_ADMIN_RPC_START_HOSTED_PROGRAM = 101;
  TAO_ADMIN_RPC_REMOVE_HOSTED_PROGRAM = 102;
  TAO_ADMIN_RPC_GET_TAO_FULL_NAME = 103;
}

message TaoAdminRequest {
  required TaoAdminRequestType rpc = 1;
  optional bytes data = 2;
  optional string path = 3;
  repeated string args = 4;
}

message TaoAdminResponse {
  required TaoAdminRequestType rpc = 1;
  required bool success = 2;
  optional bytes data = 3;
}

